Modules/_blake2/blake2s_impl.c

```diff
@@ -333,12 +333,29 @@ _blake2_blake2s_hexdigest_impl(BLAKE2sObject *self)
     return _Py_strhex((const char *)digest, self->param.digest_length);
 }
 
+/*[clinic input]
+_blake2.blake2s.__reduce__
+
+Return the digest value as a string of hexadecimal digits.
+[clinic start generated code]*/
+
+static PyObject *
+_blake2_blake2s___reduce___impl(BLAKE2sObject *self)
+/*[clinic end generated code: output=fb085f3da88f0c39 input=49898c2f1ef1ff74]*/
+{
+    PyErr_Format(PyExc_TypeError,
```

Comment:
@vstinner This function is very redundant with this issue.
Do you have any ideas with better solution?

Reply:
Maybe a public marco :)

Reply:
I don't think that exposing it to the public is the proper way.
It's should be a kind of internal utility routine.